src/rgw/rgw_admin.cc

```diff
@@ -2151,10 +2155,14 @@ int main(int argc, char **argv)
         return EINVAL;
       }
     } else if (ceph_argparse_witharg(args, i, &val, "--max-size", (char*)NULL)) {
-      max_size = (int64_t)strict_strtoll(val.c_str(), 10, &err);
+      max_size = (int64_t)strict_sistrtoll(val.c_str(), &err);
       if (!err.empty()) {
-        cerr << "ERROR: failed to parse max size: " << err << std::endl;
-        return EINVAL;
+        if (!strcmp(err.c_str(), "strict_sistrtoll: value should not be negative")) {
+          max_size = -1;
+        } else {
+          cerr << "ERROR: failed to parse max size: " << err << std::endl;
+          return EINVAL;
+        }
```

Comment:
i see that `int64_t` is not one of the template types instantiated for `strict_si_cast()` in src/common/strtol.cc, so we'll want to add that for use here.. which means we'll also need some tests for it in src/test/strtol.cc. would you be willing to look into that, or should i?